org.opentosca.bus/org.opentosca.bus.management/src/main/java/org/opentosca/bus/management/utils/MBUtils.java

```diff
@@ -141,19 +141,21 @@ public static NodeTemplateInstance getAbstractOSReplacementInstance(NodeTemplate
      *
      * @return true if the specified NodeType is one of the OperatingSystem NodeTypes. Otherwise, false.
      */
-    private static boolean isOperatingSystemNodeType(final TNodeType nodeType) {
+    private static boolean isOperatingSystemNodeType(Csar csar, final TNodeType nodeType) {
         if (nodeType.getInterfaces() == null) {
             return false;
         }
-        List<TInterface> exposedInterfaces = nodeType.getInterfaces();
-        boolean isOs = exposedInterfaces.stream()
-            .filter(tInterface -> Interfaces.OPENTOSCA_DECLARATIVE_INTERFACE_OPERATINGSYSTEM.equals(tInterface.getName()))
-            .anyMatch(os -> doesInterfaceContainOperation(os, Interfaces.OPENTOSCA_DECLARATIVE_INTERFACE_OPERATINGSYSTEM_RUNSCRIPT)
-                && doesInterfaceContainOperation(os, Interfaces.OPENTOSCA_DECLARATIVE_INTERFACE_OPERATINGSYSTEM_TRANSFERFILE));
-        boolean isDocker = exposedInterfaces.stream()
-            .filter(tInterface -> Interfaces.OPENTOSCA_DECLARATIVE_INTERFACE_DOCKERCONTAINER.equals(tInterface.getName()))
-            .anyMatch(os -> doesInterfaceContainOperation(os, Interfaces.OPENTOSCA_DECLARATIVE_INTERFACE_DOCKERCONTAINER_RUNSCRIPT)
-                && doesInterfaceContainOperation(os, Interfaces.OPENTOSCA_DECLARATIVE_INTERFACE_DOCKERCONTAINER_TRANSFERFILE));
+
+        TInterface interfaceOfNodeType = ModelUtils.getInterfaceOfNodeType(csar, nodeType, Interfaces.OPENTOSCA_DECLARATIVE_INTERFACE_OPERATINGSYSTEM);
```

Comment:
wouldn't it be more clear code if we used a method like getAllInterfacesOfNodeType() or getInheritedInterfacesOfNodeType() ?